osu.Framework/Audio/Track/TrackLoudness.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using ManagedBass;
+using ManagedBass.Loud;
+using osu.Framework.Audio.Callbacks;
+using osu.Framework.Extensions;
+using osu.Framework.Logging;
+
+namespace osu.Framework.Audio.Track
+{
+    /// <summary>
+    /// Measures loudness of audio samples.
+    /// </summary>
+    public class TrackLoudness : IDisposable
+    {
+        private Stream? data;
+
+        private readonly Task readTask;
+
+        private readonly CancellationTokenSource cancelSource = new CancellationTokenSource();
+
+        private float? integratedLoudness;
+
+        /// <summary>
+        /// Measures loudness from provided audio data.
+        /// </summary>
+        /// <param name="data">
+        /// The sample data stream.
+        /// The <see cref="TrackLoudness"/> will take ownership of this stream and dispose it when done reading track data.
+        /// If null, loudness won't get calculated.
+        /// </param>
+        public TrackLoudness(Stream? data)
+        {
+            this.data = data;
+
+            readTask = Task.Run(() =>
+            {
+                if (data == null)
+                    return;
+
+                if (Bass.CurrentDevice < 0)
+                {
+                    Logger.Log("Failed to measure loudness as no bass device is available.");
```

Comment:
I can understand logging instead of throwing here, but at the same time we should probably be logging to the error logfile
```suggestion
                    Logger.Log("Failed to measure loudness as no bass device is available.", LoggingTarget.Runtime, LogLevel.Error);
```